res/controllers/Denon-MC7000-scripts.js

```diff
@@ -1036,26 +1255,75 @@ MC7000.HotCueLED = function(value, group) {
 
 // Sampler LED
 MC7000.SamplerLED = function() {
-    for (var j = 1; j <= 4; j++) {
-        for (var i = 1; i <= 8; i++) {
+    for (var j = 0; j < 4; j++) {
+        for (var i = 0; i < 8; i++) {
+            var sampNo = 0;
             if (MC7000.PADModeSampler[j]) {
-                if (engine.getValue("[Sampler"+i+"]", "track_loaded") === 1) {
-                    if (engine.getValue("[Sampler"+i+"]", "play") === 0) {
-                        midi.sendShortMsg(0x94 + j - 1, 0x14 + i - 1, MC7000.padColor.samplerloaded);
+                if (MC7000.SamplerQty === 16) {
+                    (j >= 2) ? sampNo = (j - 2) * 8 + i + 1 : sampNo = j * 8 + i + 1;
+                } else if (MC7000.SamplerQty === 32) {
+                    sampNo = j * 8 + i + 1;
+                }
+                if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 1) {
+                    (engine.getValue("[Sampler"+sampNo+"]", "play") === 0) ? midi.sendShortMsg(0x94 + j, 0x14 + i, MC7000.padColor.samplerloaded) : midi.sendShortMsg(0x94 + j, 0x14 + i, MC7000.padColor.samplerplay);
+                } else if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 0) {
+                    midi.sendShortMsg(0x94 + j, 0x14 + i, MC7000.padColor.sampleroff);
+                }
+            }
+        }
+    }
+};
+
+// Velocity Sampler LED
+MC7000.VelSampLED = function() {
+    for (var j = 0; j < 4; j++) {
+        for (var i = 0; i < 8; i++) {
+            var sampNo = 0;
+            if (MC7000.PADModeVelSamp[j]) {
+                if (MC7000.SamplerQty === 16) {
+                    (j >= 2) ? sampNo = (j - 2) * 8 + i + 1: sampNo = j * 8 + i + 1;
```

Comment:
This is just unreadable

Reply:
>
`I guess like described here: https://support.serato.com/hc/en-us/articles/115003094007-Using-the-Pitch-Play-mode`

it is kind of similar, yes


`This is just unreadable`

I'll try to improve here

Reply:
done